core/src/main/java/org/mule/runtime/core/exception/TemplateOnErrorHandler.java

```diff
@@ -111,15 +109,11 @@ public Event process(Event event) throws MuleException {
 
         @Override
         public Publisher<Event> apply(Publisher<Event> publisher) {
-          if (flowConstruct instanceof Pipeline
-              && ((Pipeline) flowConstruct).getMessageSource() instanceof LegacyInboundEndpoint) {
-            // TODO MULE-11023 Migrate transaction execution template mechanism to use non-blocking API
-            // Use child context if HandleExceptionInterceptor is being used to avoid response being completed twice..
-            return Mono.from(publisher).flatMapMany(event -> processWithChildContext(event, p -> from(p)
-                .flatMapMany(childEvent -> Mono.from(routeAsync(childEvent, exception)))));
-          } else {
-            return from(publisher).then(event -> from(routeAsync(event, exception)));
-          }
+          // TODO MULE-11023 Migrate transaction execution template mechanism to use non-blocking API
+          // Use child context if HandleExceptionInterceptor is being used to avoid response being completed twice,
+          // and nested outer "try" blocks being skipped.
+          return Mono.from(publisher).flatMapMany(event -> processWithChildContext(event, p -> from(p)
```

Comment:
We discussed making process() blocking instead of doing this.  Just need to ensure tests pass.